doc/memos/rdm-draft-petry-riot-design-goals.md

```diff
@@ -0,0 +1,218 @@
+achievable at the expense of power consumption and configuration difficulty.
+
+## Small memory footprint
+
+RIOT aims to support constrained devices in class 1 [1]. This means that
+RIOT applications, including their libraries, consume ~ 10 KiB of RAM and ~100
+KiB of ROM, or less. All modules provided by RIOT can be used practically in
+programs of such sizes, except in exceptional circumstances.
+
+## Constrained networks
+
+Communication is typically constrained in situations where nodes are:
+
+- Distant from each other or a gateway
+- Low transmission power and separated by physical barriers such as walls.
+
+This requires physical layer protocols that can cope with high loss and low
+bandwidth.
+
+Additionally, nodes are low resource and cannot support traditional TCP/IP stack
+protocols. RIOT therefore supports emerging protocols that are the best
+available to handle these challenging conditions, as well as as a validation
+platform for them.
+
+## Short learning curve
+
+Users of RIOT want to be able to develop and deploy their products in as short a
+time as possible, with as little extra learning and specialized knowledge as
+possible. This means support for standards such as POSIX and a range of
+widely-used languages, out-of-the-box examples, simple build configuration, and
+tools and platforms that support RIOT.
+
+However, RIOT's reputation is tied to the robustness of those products, and the
+users should never be able to cut corners in such a way that the products are
+not production ready, even given tight resource constraints. 
+
+Developers, likewise, should be able to start contributing in as short a time as
+possible, and so the design and APIs should remain clear and generally follow
+common and easy to understand principles, unless there are strong benefits that
+directly achieve operational goals.
+
+## Security
+
+Security in all cases is non-negotiable and overrides any competing design
+priorities. This is because of the high importance of protecting users' privacy,
+and the corresponding risk to the emerging IoT industry. We are part of the
+solution, not the problem.
+
+Note that this use case dependent, and depends on the threats that a device may
+reasonably experience as well as the penalty if those threats are realised. RIOT
+however will cover potential shortcomings in programming or configuration in all
+use cases and assure security nonetheless.
+
+## Modularity
+
+Logical separation of the different modules of RIOT enables a highly distributed
+community to separate development work while reducing the risk of knock-on
+effects across the system. It also allows third parties to develop modules,
+expanding development efforts from just the RIOT community and repo. There is a
+balance to be struck, however: too fine-grained can come at a cost of
+unnecessary complexity, higher resources, and difficulty of understanding.
+
+## Independence of the underlying hardware
```

Comment:
Better: Hardware abstraction and cross-hardware portability

Reply:
Changed